jinja2/filters.py

```diff
@@ -567,6 +567,12 @@ def do_indent(
             rv += u'\n' + u'\n'.join(
                 indention + line if line else line for line in lines
             )
+            try:
+                # Unescape any strings that were escaped due to concatenation
+                rv = Markup(rv.unescape())
```

Comment:
This is incorrect if autoescape is currently disabled.

Reply:
Can you explain? I don't see autoescape factoring into `do_indent()` anywhere, and like I said, this is my first crack at working with the jinja2 source code. The idea of this try/escape is that we attempt to unescape, and if it's a string type instead of a `Markup` then `rv` remains unmodified. EAFP.

Reply:
OK, I think my original statement may be somewhat inaccurate, but there's an even bigger issue here. Now you're unescaping whatever came in and wrapping in in `Markup`, indicating to Jinja's autoescape mechanism that the contents are safe. So for example if I did `{{ user_input|indent }}`, and the user passed in `<big xss injection>`, your code undoes any escaping that was already done to that and tells Jinja not to escape it, so the injection gets through.